src/main/java/com/vestmark/bitbucket/plugin/MergeConflictDetectorService.java

```diff
@@ -0,0 +1,211 @@
+    {
+      int repoId = Integer.parseInt(repoIdParam);
+      long pullRequestId = Long.parseLong(pullRequestIdParam);
+      MergeConflictDetector mcd = new MergeConflictDetector(authenticationContext.getCurrentUser(), 
+        pullRequestService.getById(repoId, pullRequestId), HOST_URL);
+      BranchClassifier bc = modelService.getModel(mcd.getToRepo()).getClassifier();
+      BranchType toBranchType = bc.getType(mcd.getToBranch());
+      // If the target is not master and is a release branch, find target branch and upstream releases (if any).
+      if (!mcd.getToBranchId().equals("refs/heads/master") && toBranchType != null && toBranchType.getId().equals("RELEASE") ) {
+        //Non parallel way commented out below
+        //Page<Branch> branches = bc.getBranchesByType(toBranchType, new PageRequestImpl(0,PageRequestImpl.MAX_PAGE_LIMIT/2));
+        //branches.stream()
+        Iterable<Branch> branches = bc.getAllBranchesByType(toBranchType);
+        StreamSupport.stream(branches.spliterator(), true)
+              .filter(b -> mcd.isRelated(b))
+              .filter(b -> mcd.isUpstreamBranch(b))
+              .sorted(branchComparator)
+        //     .forEachOrdered(b -> dryRunMerge(mcd, b));
+              .forEach(b -> dryRunMerge(mcd, b));
+      }
+      dryRunMerge(mcd, refService.getDefaultBranch(mcd.getToRepo())); // Always merge to default.
+      checkForAutoMergeFailure(mcd);
+      return Response.ok(mcd.getMergeResultsModelList()).build();
+    }
+
+    private void checkForAutoMergeFailure(MergeConflictDetector mcd)
+    {
+      try {
+        PullRequestSearchRequest prsReq = new PullRequestSearchRequest.Builder().state(PullRequestState.OPEN)
+          .order(PullRequestOrder.NEWEST)
+          .fromRepositoryId(mcd.getFromRepo().getId())
+          .toRepositoryId(mcd.getToRepo().getId())
+          .build();
+        long totalNPRs = pullRequestService.count(prsReq);
+        int maxNPRs = 100;
+        if (totalNPRs < maxNPRs) {
+          maxNPRs = (int) totalNPRs;
+        }
+        int begIdx = 0;
+        while (begIdx < totalNPRs) {
+          PageRequest pgReq = new PageRequestImpl(begIdx, maxNPRs).buildRestrictedPageRequest(maxNPRs);
+          Page<PullRequest> pPRs = pullRequestService.search(prsReq, pgReq);
+          boolean amfPRExists = pPRs.stream().anyMatch(r -> r.getTitle().equals(AUTO_MERGE_FAIL));
+          if (amfPRExists) {
+            mcd.addResult(
+              refService.getDefaultBranch(mcd.getToRepo()),
+              Collections.emptyList(),
+              Arrays.asList("Please check for " + AUTO_MERGE_FAIL + "!"),
+              Collections.emptyList());
+            return;
+          }
+          begIdx += maxNPRs;
+        }
+      }
+      catch (Exception e) {
+        mcd.addResult(refService.getDefaultBranch(mcd.getToRepo()), null, Arrays.asList(e.getMessage()), null);
+      }
+    }
+
+    public void dryRunMerge(MergeConflictDetector mcd, Branch toBranch)
```

Comment:
Here's the source of the issue - mcd is mutated by the method, rather than the method returning a result object
refactor this to return a result object, then we can sort on the destination branch

Reply:
Should also be private

Reply:
Is there any reason the parallel stream above cannot but used to also check for status and then accumulate the list of results vs. having to loop multiple times?